### PR TITLE
Pkg table

### PR DESCRIPTION
**Pull request by pypingou**
These branch introduces a `packages` table, linked to the `files` table and which only stores the package name.

This should allow @charulagrl to list all the packages much more easily and quickly and thus optimize the output of the /packages/ endpoint on shumgrepper.

### CODE REVIEW

File: summershum/model.py [hunk 1]
```diff
@@ -55,7 +64,11 @@ class File(BASE):
     sha1sum = sa.Column(sa.String(40), index=True, nullable=True)
     md5sum = sa.Column(sa.String(32), index=True, nullable=True)
 
-    pkg_name = sa.Column(sa.Text, index=True, nullable=False)
+    pkg_name = sa.Column(
+        sa.Text,
+        sa.ForeignKey('Package.pkg_name', onupdate='CASCADE'),
```
Comment (puiterwijk) [inline summershum/model.py]: This should point to packages.pkg_name, given that that is the tablename of the Package class.

**Comment by ralphbean** [comment]
Any thoughts on using alembic instead of sql scripts to be run by hand?

I dream of the day we can reference an alembic ansible module, something like:

``` YAML
alembic:  state=latest config=/usr/share/summershum/alembic.ini
```

**Comment by pypingou** [comment]
I'd be +1 to use an alembic script it will just require a little more work

**Comment by ralphbean** [comment]
Cool - I won't block it if we don't want to do the alembic work now.  But it seems like we're standardizing in that direction on our other projects.

File: 0.1.6_upgrade.sql [hunk 2]
```diff
@@ -0,0 +1,13 @@
+INSERT INTO packages (pkg_name) VALUES (SELECT DISTINCT pkg_name FROM files);
```
Comment (ralphbean) [inline 0.1.6_upgrade.sql:1]: I get a syntax error in sqlite for this line.

File: summershum/model.py [hunk 3]
```diff
@@ -55,18 +90,12 @@ class File(BASE):
...
-    tarball = sa.Column(sa.Text, nullable=False)
-
-    # For now, this is an md5 handed to us by another application, so it need
-    # only be 32 bits.  But in the future we'd like to move to a more modern
-    # hash.  Therefore we size this column up to 64 bits.
-    tar_sum = sa.Column(sa.String(64), index=True, nullable=False)
-
     created_on = sa.Column(sa.DateTime, default=datetime.datetime.utcnow)
 
+    tarball = relation('Release', remote_side=[tarball], backref='files')
```
Comment (ralphbean) [inline summershum/model.py]: The `remote_side` here doesn't work -- raises an Exception.

File: 0.1.6_upgrade.sql [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+INSERT INTO releases (pkg_name, tarball, tar_sum) VALUES (
+    SELECT DISTINCT pkg_name, tar_file, tar_sum FROM files
```
Comment (ralphbean) [inline 0.1.6_upgrade.sql:4]: I don't think there is a `tar_file` column here to select.  It seems like it was renamed to `tarball` in 165ff5e04

**Comment by ralphbean** [comment]
OK.  I added alembic stuff and fixed up some other stuff.  :+1: from me.

@pypingou do you want to have another look at it and then either merge it or comment?

**Comment by pypingou** [comment]
@ralphbean forgot to push the alembic commits?

**Comment by ralphbean** [comment]
Ah, so I did.  Just pushed them in 1637f52

**Comment by pypingou** [comment]
Looks good to me :)

**Comment by ralphbean** [comment]
:fireworks: thanks! :fireworks:

**Pull request merged**